### PR TITLE
SPEC-983: Allow per-operation read concern to be used

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/SPEC-983

### CODE REVIEW

File: source/change-streams.rst [hunk 1]
```diff
@@ -228,7 +228,7 @@ The driver API consists of one helper method located on a driver’s Collection
...
 
-The aggregation command MUST use the inherited read concern of the collection it is being run against, in accordance with the `Read and Write Concern specification <https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#via-code>`_ .  The initial implementation of change streams on the server requires a “majority” read concern, drivers MUST document this requirement.  Drivers SHALL NOT throw an exception if any other read concern is specified, but instead should depend on the server to return an error.
+The aggregation command MUST use the inherited read concern of the collection it is being run against or a per-operation read concern (if supported by the driver), in accordance with the `Read and Write Concern specification <https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#via-code>`_ .  The initial implementation of change streams on the server requires a “majority” read concern, drivers MUST document this requirement.  Drivers SHALL NOT throw an exception if any other read concern is specified, but instead should depend on the server to return an error.
```
Comment (kevinAlbs) [inline source/change-streams.rst]: Since this is a MUST statement, any driver which does support per-operation read concern must make that option available in collection.watch, correct?

Comment (jmikola) [reply source/change-streams.rst]: I'd say it's still up to drivers to decide which CRUD methods support per-operation methods. CRUD alludes to the fact that methods may take those options, but doesn't address whether it's all or nothing.

The goal here was just to ensure this spec didn't prohibit the idea of per-operation options.

**Comment by ShaneHarvey** [comment]
Even though the aggregate helper MAY provide a readConcern option, the watch helper does not provide one. Are you also proposing to add a readConcern option in the ChangeStreamOptions class?

**Comment by jmikola** [comment]
> Even though the aggregate helper MAY provide a readConcern option, the watch helper does not provide one. Are you also proposing to add a readConcern option in the ChangeStreamOptions class?

In "Where is read concern?" within the [CRUD spec Q&A](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#q-a), the text merely says that drivers may provide an option. No where in the spec is `readPreference`, `readConcern`, or `writeConcern` mentioned in any method's options class. I'm not proposing we modify this spec in any way beyond what I have in the PR. I'd just like to remove exclusive language that says driver's must use the Collection's read concern and nothing else.

File: source/change-streams.rst (hunk 1 above)
Comment (kevinAlbs) [reply source/change-streams.rst]: Ah ok. Up to you, but maybe reword slightly to clarify that supporting the per-operation read concern isn't necessary even if other CRUD methods support it. Something along the lines of "The aggregation command MUST use the inherited read concern of the collection it is being run against. This may be overridden by a per-operation read concern if supported by the driver."

**Comment by kevinAlbs** [comment]
LGTM with optional rewording

File: source/change-streams.rst [hunk 2]
```diff
@@ -228,7 +228,7 @@ The driver API consists of one helper method located on a driver’s Collection
...
 
-The aggregation command MUST use the inherited read concern of the collection it is being run against, in accordance with the `Read and Write Concern specification <https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#via-code>`_ .  The initial implementation of change streams on the server requires a “majority” read concern, drivers MUST document this requirement.  Drivers SHALL NOT throw an exception if any other read concern is specified, but instead should depend on the server to return an error.
+The aggregation command MUST use the inherited read concern of the collection it is being run against, in accordance with the `Read and Write Concern specification <https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#via-code>`_.  This MAY be overridden by a per-operation read concern if supported by the driver.  The initial implementation of change streams on the server requires a “majority” read concern, drivers MUST document this requirement.  Drivers SHALL NOT throw an exception if any other read concern is specified, but instead should depend on the server to return an error.
```
Comment (jmikola) [inline source/change-streams.rst]: Per @kevinAlbs' suggestion, I left the original text as-is and just added a sentence inbetween:

> This MAY be overridden by a per-operation read concern if supported by the driver.

Comment (mbroadst) [reply source/change-streams.rst]: Maybe the wording should really just be "The aggregation command MUST use the rules for read concerns documented in the Read and Write concern specification..." since the language there already contains "In addition, a driver MAY allow it to be specified on a per-operation basis in accordance with the CRUD specification." ?

Comment (jmikola) [reply source/change-streams.rst]: Done. See updated commit.
